### PR TITLE
Ported to OM 23_12

### PR DESCRIPTION
**Pull request by s7design**

### CODE REVIEW

File: Kernel/Modules/AdminPGP.pm [hunk 1]
```diff
@@ -21,13 +21,6 @@ sub new {
...
     bless( $Self, $Type );
 
-    # check all needed objects
-    for (qw(ParamObject DBObject LayoutObject ConfigObject LogObject MainObject EncodeObject)) {
-        if ( !$Self->{$_} ) {
-            $Self->{LayoutObject}->FatalError( Message => "Got no $_!" );
-        }
-    }
-
     $Self->{CryptObject} = Kernel::System::Crypt->new( %Param, CryptType => 'PGP' );
```
Comment (moritz) [inline Kernel/Modules/AdminPGP.pm]: The correct way to do that (IMHO) is

     $Kernel::OM->ObjectParamAdd(
        'Kernel::System::Crypt' => {
            CryptType => 'PGP',
       }
     );
     $Self->{CryptObject} = $Kernel::OM->Get('Kernel::System::Crypt');

Comment (s7design) [inline Kernel/Modules/AdminPGP.pm]: Thanks for follow-up.
We will correct  this as you suggested. 

Regards 
Zoran

**Comment by mrcbnsls** [comment]
@mgruner, lets talk about this one once you're back.

**Comment by s7design** [comment]
I have deleted AdminPGP form this PR, I will fix them in another PR

Regards
Zoran

**Comment by mgruner** [comment]
Merged, :+1:

**Pull request closed without merging**